src/main/java/liquibase/ext/spanner/snapshotgenerator/SequenceSnapshotGeneratorSpanner.java

```diff
@@ -0,0 +1,74 @@
+package liquibase.ext.spanner.snapshotgenerator;
+
+import liquibase.database.Database;
+import liquibase.ext.spanner.ICloudSpanner;
+import liquibase.snapshot.SnapshotGenerator;
+import liquibase.snapshot.jvm.SequenceSnapshotGenerator;
+import liquibase.statement.SqlStatement;
+import liquibase.statement.core.RawParameterizedSqlStatement;
+import liquibase.structure.DatabaseObject;
+import liquibase.structure.core.ForeignKey;
+import liquibase.structure.core.Schema;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class SequenceSnapshotGeneratorSpanner extends SequenceSnapshotGenerator {
+
+  /**
+   * This generator will be in all chains relating to CloudSpanner, whether or not
+   * the objectType is {@link liquibase.structure.core.Sequence}.
+   */
+  @Override
+  public int getPriority(Class<? extends DatabaseObject> objectType, Database database) {
+    if (database instanceof ICloudSpanner) {
+      return PRIORITY_DATABASE;
+    }
+    return PRIORITY_NONE;
+  }
+
+  @Override
+  protected SqlStatement getSelectSequenceStatement(Schema schema, Database database) {
+    if (database instanceof ICloudSpanner) {
+      List<String> parameter = new ArrayList<>(2);
+      parameter.add(database.getDefaultCatalogName());
+      parameter.add(database.getDefaultSchemaName());
+
+      StringBuilder sql = new StringBuilder("SELECT seq.NAME AS SEQUENCE_NAME, ")
+                              .append("seq_kind.OPTION_VALUE AS SEQUENCE_KIND, ")
+                              .append("skip_max.OPTION_VALUE AS SKIP_RANGE_MAX, ")
+                              .append("skip_min.OPTION_VALUE AS SKIP_RANGE_MIN, ")
+                              .append("start_counter.OPTION_VALUE AS START_VALUE ")
+                              .append("FROM (SELECT DISTINCT NAME ")
+                              .append("FROM INFORMATION_SCHEMA.sequence_options ")
+                              .append("WHERE CATALOG = ? AND SCHEMA = ?) AS seq ")
+                              .append("LEFT JOIN INFORMATION_SCHEMA.sequence_options AS seq_kind ")
+                              .append("ON seq.NAME = seq_kind.NAME ")
```

Comment:
All these join clauses should be:

`ON seq.catalog=seq_kind.catalog AND seq.schema=seq_kind.schema AND seq.name = seq_kind.name`

In the current query, things will go wrong if anyone creates two different sequences with the same name in two different schemas.

Reply:
fixed